File: compiler/src/main/java/edu/tum/ase/compiler/model/SourceCode.java
package edu.tum.ase.compiler.model;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
public class SourceCode {
    private String code;
    private String fileName;
    private String stdout;
    private String stderr;
    private boolean compilable = false;
}

File: frontend code/src/app/model/sourceCode.ts
export class SourceCode {
  code: string;
  fileName: string;
  stdout: string;
  stderr: string;
  compilable: boolean;

  constructor(code: string, fileName: string) {
    this.code = code;
    this.fileName = fileName;
    this.stdout = '';
    this.stderr = '';
    this.compilable = false;
  }
}
